module/zfs/spa.c

```diff
@@ -3737,6 +3746,61 @@ spa_activity_check(spa_t *spa, uberblock_t *ub, nvlist_t *config)
 	return (error);
 }
 
+/*
+ * Called from zfs_ioc_clear for a pool that was suspended
+ * after failing mmp write checks.
+ */
+boolean_t
+spa_mmp_remote_host_activity(spa_t *spa)
+{
+	ASSERT(spa_multihost(spa) && spa_suspended(spa));
+
+	nvlist_t *best_label;
+	uberblock_t best_ub;
+
+	/*
+	 * Locate the best uberblock on disk
+	 */
+	vdev_uberblock_load(spa->spa_root_vdev, &best_ub, &best_label);
+	if (best_label) {
+		/*
+		 * confirm that the best hostid matches our hostid
+		 */
+		if (nvlist_exists(best_label, ZPOOL_CONFIG_HOSTID) &&
```

Comment:
If the best_label we found does not have an nvpair for hostid, should we exit with B_TRUE?

Reply:
Yes, I think so.  This state should be impossible, but if the `best_label` somehow doesn't have a `ZPOOL_CONFIG_HOSTID` then we want to err on the side of caution and not attempt to resume the pool.